mozdownload/scraper.py

```diff
@@ -451,6 +451,8 @@ def get_latest_build_date(self):
                           months.entries[-1] + '/')
             parser = self._create_directory_parser(url)
             parser.entries = parser.filter(r'.*%s' % self.platform_regex)
+            if not parser.entries:
+                raise errors.NotSupportedError('No builds have been found')
```

Comment:
I just noticed that this should be a `NotFoundError`. Looks like it slipped through in my last review. Can you please update both the code here and the test? 

Beside that we probably should indeed have a `NotSupportedError` thrown when an unsupported platform has been specified. But this check should be done as early as possible when we know the application and which kind of platform is selected. I would suggest that you create a separate PR for that and we leave open the issue even with this PR merged.